### PR TITLE
Support * event type for mapping/handling

### LINKED ISSUE #10: Support * event type for handlers and mappers
It can be useful to map or handle all events. For example,
an event logger that logs all events. We can support this
through an interface such as:

```js
events.on('*', payload => {});
events.map('*', payload => {});
```

### PR DESCRIPTION
**Pull request by ganemone**
Fixes #10

### CODE REVIEW

File: src/emitter.js [hunk 1]
```diff
@@ -41,8 +42,10 @@ export default class UniversalEmitter {
...
   emit(type, payload, ctx) {
-    const mappers = this.mappers[type] || [];
-    const handlers = this.handlers[type] || [];
+    const globalMappers = this.mappers[globalEventType] || [];
+    const globalHandlers = this.handlers[globalEventType] || [];
+    const mappers = (this.mappers[type] || []).concat(globalMappers);
+    const handlers = (this.handlers[type] || []).concat(globalHandlers);
     const event = {
       type,
       payload: mappers.reduce((payload, mapper) => {
```
Comment (AlexMSmithCA) [inline src/emitter.js]: It looks like we previously had an implicit wildcard match in `getArgs` below, which mapped to all key-less types.  Given this change, that would no longer be supported (a la 'all key-less types are handled together').  Is that something we wanted to support?

Comment (ganemone) [reply src/emitter.js]: I think the intended behavior of that was to have * behave as a wildcard. I don't think there is any value in supporting events with no type.

**Comment by AlexMSmithCA** [comment]
I agree it may potentially be misleading for folks (_Cool!  I can wildcard anything... myPlugin::*_) although I will say the use-case itself seems very reasonable.  As an alternative, I could see expanding the surface area of the API with something like _.onAny_ or _.onGlobal_

**Comment by ganemone** [comment]
> Listening on * seems like it could be a footgun to me. If we're sure though...

It definitely is. tbh, I would consider having this be an undocumented feature. We are only adding it because we need it to accomplish something somewhat uber specific.

**Comment by ganemone** [comment]
!merge

**Pull request merged**